kombu/transport/SQS.py

```diff
@@ -200,10 +201,10 @@ def _extract_backoff_policy_configuration_and_message(self, delivery_tag):
         return routing_key, message, backoff_tasks, backoff_policy
 
     def apply_backoff_policy(self, routing_key, delivery_tag,
-                             backoff_policy, backoff_tasks):
+                             backoff_policy, backoff_tasks, message):
         queue_url = self.channel._queue_cache[routing_key]
         task_name, number_of_retries = \
-            self.extract_task_name_and_number_of_retries(delivery_tag)
+            self.extract_task_name_and_number_of_retries(message)
```

Comment:
Shouldn't we get the message using `message = self._delivered[delivery_tag]`?
This PR breaks the class' public API and I'd rather avoid doing that.